### PR TITLE
Update the pull request template

### PR DESCRIPTION
**Pull request by stephanie-wang**
## What do these changes do?

Previously, when opening a PR, the primary question was about what changes were included. This updates the github PR template to add a "Why" question, which should make it clearer that the contributor should also give some context for the problem that the PR addresses.

## Linter

- [x] I've run `scripts/format.sh` to lint the changes in this PR.

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -4,6 +4,10 @@ Thank you for your contribution!
 Please review https://github.com/ray-project/ray/blob/master/CONTRIBUTING.rst before opening a pull request.
 -->
 
+## Why are these changes needed?
+
+<!-- Please give a short brief about the problem these changes address. -->
```
Comment (edoakes) [inline .github/PULL_REQUEST_TEMPLATE.md]:
```suggestion
<!-- Please give a short summary of the problem these changes address. -->
```

This is not directly related to your change, but while we're at it might be good to change this wording. "Brief" as a noun is not a very common word so it could be less understandable (especially to non-native English speakers).

Also, "short brief" is pretty redundant...

Comment (edoakes) [reply .github/PULL_REQUEST_TEMPLATE.md]: Applies to the "what" section below as well

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -4,6 +4,10 @@ Thank you for your contribution!
 Please review https://github.com/ray-project/ray/blob/master/CONTRIBUTING.rst before opening a pull request.
 -->
 
+## Why are these changes needed?
+
+<!-- Please give a short summary of the problem these changes address. -->
+
 ## What do these changes do?
 
 <!-- Please give a short brief about these changes. -->
```
Comment (stephanie-wang) [inline .github/PULL_REQUEST_TEMPLATE.md]:
```suggestion
<!-- Please give a short summary about these changes. -->
```

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 3]
```diff
@@ -4,9 +4,13 @@ Thank you for your contribution!
...
 -->
 
+## Why are these changes needed?
+
+<!-- Please give a short summary of the problem these changes address. -->
+
 ## What do these changes do?
 
-<!-- Please give a short brief about these changes. -->
+<!-- Please give a short summary about these changes. -->
```
Comment (robertnishihara) [inline .github/PULL_REQUEST_TEMPLATE.md]:
```suggestion
<!-- Please give a short summary of these changes. -->
```

**Review by robertnishihara — APPROVED** [review]
